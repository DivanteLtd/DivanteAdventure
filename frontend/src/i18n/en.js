export default {
  global_search: {
    tribe_url: 'Tribe page',
    project_url: 'Project page',
  },
  date: {
    firstDayOfWeek: '0',
    months: [
      'January',
      'February',
      'March',
      'April',
      'May',
      'June',
      'July',
      'August',
      'September',
      'October',
      'November',
      'December',
    ],
    formats: {
      hourDuration: '{HH} h',
      shortDate: 'MMMM D, YYYY',
      dateWithWeekDay: 'dddd, MMMM D YYYY',
    },
  },
  snackbar_redirect: {
    agreements: 'Before starting using Adventure accept all required agreements concerning General Data Protection Regulation',
    edit: 'Before starting using Adventure fill all required form fields',
  },
  menu: {
    group: {
      administration: 'Administration',
      adventure: 'Adventure',
      agreements: 'Agreements',
      systemHR: 'HR',
      planner: 'Scheduler',
    },
    entry: {
      dashboard: 'Dashboard',
      feedback: 'Feedback',
      personsList: 'Person\'s list',
      evidences: 'Evidences',
      overtime: 'Overtime',
      agreements_general: 'General',
      agreements_marketing: 'Marketing',
      agreements_acceptations: 'Acceptation list',
      firm: 'Firm',
      firm_departments: 'Departments',
      firm_employees: "Person's",
      firm_projects: 'Projects',
      firm_tribes: 'Practice and departments',
      free_days: 'Free days',
      unavailability_days: 'Unavailability days',
      leave_days: 'Leave days',
      free_days_dashboard: 'Free days dashboard',
      gdpr: 'GDPR',
      checklists: 'Checklists',
      general: 'General',
      checklistTemplates: 'Checklist templates',
      requests: 'Requests',
      scheduler: 'Scheduler',
      rotationStats: 'Rotation statistics',
      positions: 'Managing positions',
      monthlyReport: 'Monthly report',
      tribeStats: 'Practice stats',
      faq: 'FAQ',
      scheduler_stats: 'Scheduler - stats',
      hardware_agreements: 'Hardware lending agreements',
      system_config: 'Configuration',
      system_dictionaries: 'Dictionaries',
    },
  },
  config: {
    groups: {
      slack: 'Slack',
      gitlab: 'Gitlab',
      snipe_it: 'Snipe IT',
      avaza: 'Avaza',
      tribe: 'Connected tribes',
      content: 'Content',
    },
    entries: {
      slack: {
        client_id: 'Client ID',
        client_secret: 'Client secret',
        redirect_uri: 'Redirect URL address',
        admin_id: 'Admin integration ID',
        admin_access_token: 'Admin integration access token',
        admin_status: 'Admin integration status',
      },
      gitlab: {
        instance_url: 'Instance URL address',
        token: 'Access token',
      },
      snipe_it: {
        instance_url: 'Instance URL address',
        token: 'Access token',
      },
      avaza: {
        token: 'Access token',
        sick_leave_project_id: 'Project ID for sick leave',
        sick_leave_category_id: 'Timesheet category ID for sick leave',
        free_day_project_id: 'Project ID for free day',
        free_day_category_id: 'Timesheet category ID for free day',
      },
      tribe: {
        connected1_id: 'Tribe connected 1 ID',
        connected2_id: 'Tribe connected 2 ID',
      },
      content: {
        agreement_general_en: 'Agreements - general info EN',
        agreement_ohs_en: 'Agreements - OHS info EN',
        agreement_gdpr_en: 'Agreements - GDPR info EN',
        agreement_iso_en: 'Agreements - ISO info EN',
        agreement_marketing_main_en: 'Agreements - Marketing info main EN',
        agreement_marketing_en: 'Agreements - Marketing info EN',
        iso_link_en: 'ISO Management System link EN',
        agreement_general_pl: 'Agreements - general info PL',
        agreement_ohs_pl: 'Agreements - OHS info PL',
        agreement_gdpr_pl: 'Agreements - GDPR info PL',
        agreement_iso_pl: 'Agreements - ISO info PL',
        agreement_marketing_main_pl: 'Agreements - Marketing info main PL',
        agreement_marketing_pl: 'Agreements - Marketing info PL',
        iso_link_pl: 'ISO Management System link PL',
        email_domain: 'Domain of created employee email',
        work_mode_link: 'Work mode - Confluence link',
      },
    },
    freeDays: {
      christian: {
        easter: 'Easter (Christian)',
        easter_monday: 'Easter Monday (Christian)',
        corpus_christi: 'Corpus Christi (Christian)',
        pentecost: 'Pentecost (Christian)',
        good_friday: 'Good Friday (Christian)',
        ascension_day: 'Ascension Day (Christian)',
        pentecost_monday: 'Pentecost Monday (Christian)',
      },
      us: {
        martin_luther_king_jr_birthday: 'Martin Luther King Jr. Day (USA)',
        george_washington_birthday: 'George Washington Day (USA)',
        memorial_day: 'Memorial Day (USA)',
        labor_day: 'Labor Day (USA)',
        columbus_day: 'Columbus Day (USA)',
        thanksgiving_day: 'Thanksgiving Day (USA)',
      },
    },
  },
  notifications: {
    'You have been added to tribe': 'You have been added to tribe',
    'Your request is accepted ID:': 'Your request is accepted ID:',
    'Your resignation request is accepted ID:': 'Your resignation request is accepted ID:',
    'Your resignation request is rejected ID:': 'Your resignation request is rejected ID:',
    'Your request is rejected ID:': 'Your request is rejected ID:',
    'You have outstanding request to manage for': 'You have outstanding request to manage for',
    'You have outstanding planned request to accept for:': 'You have outstanding planned request to accept for:',
    'You have outstanding evidence to manage for': 'You have outstanding evidence to manage for',
    'Period has been edited in your project': 'Period has been edited in your project',
    'Person has resign from resignation request ID:': 'Person has resign from resignation request ID:',
    'Person has resign from request ID:': 'Person has resign from request ID:',
    'Person has resign from planned request ID:': 'Person has resign from planned request ID:',
    'You have outstanding resign request to manage for': 'You have outstanding resign request to manage for',
    'All yours periods have been deleted from project': 'All yours periods have been deleted from project',
    'One period have been deleted from your project': 'One period have been deleted from your project',
    'You have been deleted from tribe': 'You have been deleted from tribe',
    'You have outstanding hardware lending agreement to generate for': 'You have outstanding hardware lending agreement to generate for',
    'You have outstanding hardware lending agreement to sign for': 'You have outstanding hardware lending agreement to sign for',
    'You have been given a new period in the project': 'You have been given a new period in the project',
    'Logout successfully': 'Logout successfully',
  },
};
