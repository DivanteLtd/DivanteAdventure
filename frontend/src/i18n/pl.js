export default {
  global_search: {
    tribe_url: 'Strona plemienia',
    project_url: 'Strona projektu',
  },
  date: {
    firstDayOfWeek: '1',
    months: [
        'Styczeń',
        'Luty',
        'Marzec',
        'Kwiecień',
        'Maj',
        'Czerwiec',
        'Lipiec',
        'Sierpień',
        'Wrzesień',
        'Październik',
        'Listopad',
        'Grudzień',
    ],
    formats: {
      hourDuration: '{HH} godz.',
      shortDate: 'D MMMM YYYY',
      dateWithWeekDay: 'dddd, D MMMM YYYY',
    },
  },
  snackbar_redirect: {
    agreements: 'Przed rozpoczęciem korzystania z Adventure zaakceptuj wszystkie wymagane zgody',
    edit: 'Przed rozpoczęciem korzystania z Adventure uzupełnij wszystkie wymagane pola formularza',
  },
  menu: {
    group: {
      administration: 'Administracja',
      adventure: 'Adventure',
      agreements: 'Zgody',
      systemHR: 'HR',
      planner: 'Planer',
    },
    entry: {
      dashboard: 'Dashboard',
      feedback: 'Feedback',
      evidences: 'Ewidencje',
      overtime: 'Nadgodziny',
      personsList: 'Lista osób',
      agreements_general: 'Ogólne',
      agreements_marketing: 'Marketingowe',
      agreements_acceptations: 'Lista akceptacji',
      firm: 'Firma',
      firm_departments: 'Działy',
      firm_employees: 'Osoby',
      firm_projects: 'Projekty',
      firm_tribes: 'Praktyki i działy',
      free_days: 'Dni wolne',
      unavailability_days: 'Dni niedostępności',
      leave_days: 'Dni urlopowe',
      free_days_dashboard: 'Podsumowanie dni wolnych',
      general: 'Ogólne',
      checklistTemplates: 'Szablony checklist',
      requests: 'Wnioski',
      scheduler: 'Planer',
      rotationStats: 'Statystyki rotacji',
      positions: 'Zarządzanie stanowiskami',
      monthlyReport: 'Raport miesięczny',
      tribeStats: 'Statystyki praktyk',
      checklists: 'Checklisty',
      faq: 'FAQ',
      scheduler_stats: 'Planer - statystyki',
      hardware_agreements: 'Umowy użyczenia sprzętu',
      system_config: 'Konfiguracja',
      system_dictionaries: 'Słowniki',
    },
  },
  config: {
    groups: {
      slack: 'Slack',
      gitlab: 'Gitlab',
      snipe_it: 'Snipe IT',
      avaza: 'Avaza',
      tribe: 'Połączone praktyki',
      content: 'Treści',
      payroll: 'Płace i kadry',
      evidence: 'Ewidencja',
    },
    entries: {
      slack: {
        client_id: 'ID klienta',
        client_secret: 'Sekret klienta',
        redirect_uri: 'Adres przekierowania',
        admin_id: 'ID integracji admina',
        admin_access_token: 'Token dostępu integracji admina',
        admin_status: 'Status integracji admina',
      },
      gitlab: {
        instance_url: 'Adres URL instancji',
        token: 'Token dostępu',
      },
      snipe_it: {
        instance_url: 'Adres URL instancji',
        token: 'Token dostępu',
      },
      avaza: {
        token: 'Token dostępu',
        sick_leave_project_id: 'ID projektu zwolnienia chorobowego',
        sick_leave_category_id: 'ID kategorii zwolnienia chorobowego',
        free_day_project_id: 'ID projektu dnia wolnego',
        free_day_category_id: 'ID kategorii dnia wolnego',
      },
      tribe: {
        connected1_id: 'Połączone plemię 1 ID',
        connected2_id: 'Połączone plemię 2 ID',
      },
      content: {
        agreement_general_en: 'Zgody - ogólne info EN',
        agreement_ohs_en: 'Zgody - BHP info EN',
        agreement_gdpr_en: 'Zgody - RODO info EN',
        agreement_iso_en: 'Zgody - ISO info EN',
        agreement_marketing_main_en: 'Zgody - Marketingowe info główne EN',
        agreement_marketing_en: 'Zgody - Marketingowe info EN',
        iso_link_en: 'Link do Zarządzania Systemem ISO EN',
        agreement_general_pl: 'Zgody - ogólne info PL',
        agreement_ohs_pl: 'Zgody - BHP info PL',
        agreement_gdpr_pl: 'Zgody - RODO info PL',
        agreement_iso_pl: 'Zgody - ISO info PL',
        agreement_marketing_main_pl: 'Zgody - Marketingowe info główne PL',
        agreement_marketing_pl: 'Zgody - Marketingowe info PL',
        iso_link_pl: 'Link do Zarządzania Systemem ISO PL',
        email_domain: 'Domena utworzonego adresu e-mail pracownika',
        work_mode_link: 'Tryb pracy - link do Confluence',
      },
      payroll: {
        email_for_b2b_pl: 'Adres (adresy) email do osoby (osób) zajmujących się pracownikami (B2B) mieszkającymi na terenie Polski. Możliwość wprowadzenia kilka adresów po przecinku',
        email_for_b2b_de: 'Adres (adresy) email do osoby (osób) zajmujących się pracownikami (B2B) mieszkającymi na terenie Niemczech. Możliwość wprowadzenia kilka adresów po przecinku',
        email_for_not_b2b_pl: 'Adres (adresy) email do osoby (osób) zajmujących się pracownikami (inne niż B2B) mieszkającymi na terenie Polski. Możliwość wprowadzenia kilka adresów po przecinku',
        email_for_not_b2b_de: 'Adres (adresy) email do osoby (osób) zajmujących się pracownikami (inne niż B2B) mieszkającymi na terenie Niemczech. Możliwość wprowadzenia kilka adresów po przecinku',
        email_for_coe_pl: 'Adres (adresy) email do osoby (osób) zajmujących się pracownikami (Umowa o pracę) mieszkającymi na terenie Polski. Możliwość wprowadzenia kilka adresów po przecinku',
        email_for_coe_de: 'Adres (adresy) email do osoby (osób) zajmujących się pracownikami (Umowa o pracę) mieszkającymi na terenie Niemczech. Możliwość wprowadzenia kilka adresów po przecinku',
      },
      evidence: {
        email_for_b2b_pl: 'Adresy email do osób zajmujących się fakturami kosztowymi pracowników (B2B) mieszkającymi na terenie Polski. Możliwość wprowadzenia kilka adresów po przecinku',
        email_for_b2b_de: 'Adresy email do osób zajmujących się fakturami kosztowymi pracowników (B2B) mieszkającymi na terenie Niemczech. Możliwość wprowadzenia kilka adresów po przecinku',
        email_for_not_b2b_pl: 'Adresy email do osób zajmujących się fakturami kosztowymi pracowników (inne niż B2B) mieszkającymi na terenie Polski. Możliwość wprowadzenia kilka adresów po przecinku',
        email_for_not_b2b_de: 'Adresy email do osób zajmujących się fakturami kosztowymi pracowników (inne niż B2B) mieszkającymi na terenie Niemczech. Możliwość wprowadzenia kilka adresów po przecinku',
      },
    },
    freeDays: {
      christian: {
        easter: 'Wielkanoc (chrześcijański)',
        easter_monday: 'Poniedziałek Wielkanocny (chrześcijański)',
        corpus_christi: 'Boże Ciało (chrześcijański)',
        pentecost: 'Zielone Świątki (chrześcijański)',
        good_friday: 'Wielki Piątek (chrześcijański)',
        ascension_day: 'Wniebowstąpienie (chrześcijański)',
        pentecost_monday: 'Drugi dzień Zielonych Świątek (chrześcijański)',
      },
      us: {
        martin_luther_king_jr_birthday: 'Dzień Martina Luthera Kinga Jr. (USA)',
        george_washington_birthday: 'Dzień Jerzego Waszyngtona (USA)',
        memorial_day: 'Memorial Day (USA)',
        labor_day: 'Święto Pracy (USA)',
        columbus_day: 'Dzień Kolumba (USA)',
        thanksgiving_day: 'Dzień Dziękczynienia (USA)',
      },
    },
  },
  notifications: {
    'You have been added to tribe': 'Zostałeś dodany do plemienia',
    'Your request is accepted ID:': 'Twój wniosek jest zaakceptowany ID:',
    'Your resignation request is accepted ID:': 'Twój wniosek o rezygnacje jest zaakceptowany ID:',
    'Your resignation request is rejected ID:': 'Twój wniosek o rezygnacje zastał odrzucony ID:',
    'Your request is rejected ID:': 'Twój wniosek został odrzucony ID:',
    'You have outstanding request to manage for': 'Masz oczekujący wniosek do rozpatrzenia dla',
    'You have outstanding planned request to accept for:': 'Masz oczekujący zaplanowany wniosek do akceptacji dla',
    'You have outstanding evidence to manage for': 'Masz oczekującą ewidencję do rozpatrzenia dla',
    'Period has been edited in your project': 'Został zmieniony Twój okres w projekcie',
    'Person has resign from resignation request ID:': 'Osoba anulowała swoją rezygnację z wniosku ID:',
    'Person has resign from request ID:': 'Osoba zrezygnowała z wniosku ID:',
    'Person has resign from planned request ID:': 'Osoba zrezygnowała z zaplanowanego wniosku ID:',
    'You have outstanding resign request to manage for': 'Masz oczekujący wniosek o rezygnacje do rozpatrzenia dla',
    'All yours periods have been deleted from project': 'Usunięto wszystkie Twoje okresy z projektu',
    'One period have been deleted from your project': 'Usunięto Ci okres w projekcie',
    'You have been deleted from tribe': 'Zostałeś usunięty z plemienia',
    'You have outstanding hardware lending agreement to generate for': 'Masz oczekującą umowę użyczenia sprzętu do wygenerowania dla',
    'You have outstanding hardware lending agreement to sign for': 'Masz oczekującą umowę użyczenia sprzętu do podpisania dla',
    'You have been given a new period in the project': 'Przydzielono Ci nowy okres w projekcie',
    'Logout successfully': 'Wylogowano poprawnie',
  },
};
