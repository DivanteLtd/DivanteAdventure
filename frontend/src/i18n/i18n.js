import Vue from 'vue';
import VueI18n from 'vue-i18n';
import pl from './pl';
import en from './en';
import moment from '@divante-adventure/work-moment';

const messages = {
  pl,
  en,
};

export const languages = [
  {
    text: 'English / Angielski',
    value: 'en',
  }, {
    text: 'Polish / Polski',
    value: 'pl',
  },
];

export const supportedLanguages = [
  {
    code: 'pl',
    label: {
      pl: 'Polski',
      en: 'Polish',
    },
  }, {
    code: 'en',
    label: {
      pl: 'Angielski',
      en: 'English',
    },
  },
];

const splittedMonths = {
  pl: [
    'Sty.czeń',
    'Lu.ty',
    'Ma.rzec',
    'Kwie.cień',
    'Maj',
    'Czer.wiec',
    'Li.piec',
    'Sier.pień',
    'Wrze.sień',
    'Paź.dzier.nik',
    'Lis.to.pad',
    'Gru.dzień',
  ],
  en: [
    'Ja.nu.a.ry',
    'Feb.ru.a.ry',
    'March',
    'Ap.ril',
    'May',
    'June',
    'Ju.ly',
    'Au.gust',
    'Sep.tem.ber',
    'Oc.to.ber',
    'No.vem.ber',
    'De.cem.ber',
  ],
};


const fallbackLanguage = 'en';

export function getSuggestedLanguage() {
  return localStorage.getItem('chosenLanguage') || fallbackLanguage;
}

export function setLanguage(languageCode) {
  const supportedCodes = supportedLanguages.map(lang => lang.code);
  if (supportedCodes.includes(languageCode)) {
    localStorage.setItem('chosenLanguage', languageCode);
    window.location.reload();
  }
}

export function isSupportedLanguage(languageCode) {
  const supportedCodes = supportedLanguages.map(lang => lang.code);
  return supportedCodes.includes(languageCode);
}


Vue.use(VueI18n);
const i18n = new VueI18n({
  locale: getSuggestedLanguage(),
  fallbackLocale: fallbackLanguage,
  messages,
  silentTranslationWarn: true,
});

export function getSplittedMonth(month) {
  return splittedMonths[getSuggestedLanguage()][month];
}

export function reportHeaders(header) {
  if (getSuggestedLanguage() === 'pl') {
    switch (header) {
      case 'person':
        return 'osoba';
      case 'project':
        return 'projekt';
      case 'work time':
        return 'etat';
      case 'available time':
        return 'dostępny czas';
      case 'planned':
        return 'zaplanowano';
      default:
        return null;
    }
  } else {
    return header;
  }
}

export function getShortWeekday(weekday) {
  if (getSuggestedLanguage() === 'pl') {
    return (['Ni', 'Pn', 'Wt', 'Śr', 'Cz', 'Pt', 'So'])[weekday];
  } else {
    return (['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'])[weekday];
  }
}

export {
  i18n,
};

moment.locale(getSuggestedLanguage());
