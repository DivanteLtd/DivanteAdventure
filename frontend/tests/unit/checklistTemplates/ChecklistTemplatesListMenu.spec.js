import { shallowMount, mount } from '@vue/test-utils';
import ChecklistTemplatesListMenu
    from '../../../src/components/hr/checklistTemplates/checklistTemplatesView/ListMenu.vue';
import Vuetify from 'vuetify';
import Vue from 'vue';
import i18n from '../../../src/i18n/i18n';

Vue.use(Vuetify);

function createConfig(overrides) {
    const mocks = {
        props: {
            visible: false,
        },
        i18n,
    };
    return Object.assign({ mocks }, overrides);
}

describe('ChecklistTemplatesCard.vue', () => {
    test('selector exists', () => {
        const config = createConfig();
        const wrapper = shallowMount(ChecklistTemplatesListMenu, config);
        expect(wrapper.find('#checklist-templates-list-menu').exists()).toBeTruthy();
    });
    test('renders correctly', () => {
        const config = createConfig();
        const wrapper = shallowMount(ChecklistTemplatesListMenu, config);
        expect(wrapper.element).toMatchSnapshot();
    });
    test('dialog dont exist, visible set to false', () => {
        const config = createConfig();
        const wrapper = shallowMount(ChecklistTemplatesListMenu, config);
        expect(wrapper.find('#checklist-templates-list-menu').exists()).toBeTruthy();
    });
    test('dialog visible correctly, visible set to true', () => {
        const wrapper = mount(ChecklistTemplatesListMenu, { mocks: {
            props: {
                    visible: true,
            },
        } });
        expect(wrapper.find('#loading-dialog').exists()).toBeTruthy();
    });
});
