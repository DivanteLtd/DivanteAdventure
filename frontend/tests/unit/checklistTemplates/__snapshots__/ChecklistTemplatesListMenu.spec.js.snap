// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChecklistTemplatesCard.vue renders correctly 1`] = `
<div
  id="checklist-templates-list-menu"
>
  <v-menu-stub
    closedelay="0"
    closeonclick="true"
    closeoncontentclick="true"
    contentclass=""
    maxheight="auto"
    maxwidth="auto"
    nudgebottom="0"
    nudgeleft="0"
    nudgeright="0"
    nudgetop="0"
    nudgewidth="0"
    offsety="true"
    opendelay="0"
    openonclick="true"
    origin="top left"
    transition="v-menu-transition"
  >
     
    <v-list-stub
      tag="div"
      tile="true"
    >
      <v-list-item-stub
        activeclass=""
        tag="div"
      >
        <v-list-item-title-stub>
          Add new checklist template
        </v-list-item-title-stub>
      </v-list-item-stub>
    </v-list-stub>
  </v-menu-stub>
   
  <loading-dialog-stub />
</div>
`;
